src/Component/RuleTable/RuleTable.tsx

```diff
@@ -104,6 +113,85 @@ export class RuleTable extends React.Component<RuleTableProps, RuleTableState> {
     rendererType: 'OpenLayers'
   };
 
+  static getDerivedStateFromProps(nextProps: RuleTableProps, prevState: RuleTableState) {
+    let countsAndDuplicates;
+    try {
+      let filtersEqual = true;
+      nextProps.rules.forEach((rule, index) => {
+        try {
+          filtersEqual = filtersEqual && _isEqual(prevState.rules[index].filter, rule.filter);
+        } catch (e) {
+          filtersEqual = false;
+        }
+      });
+      // we should refrain from using _isEqual on the data as it's very slow on largish datasets
+      // thus it's the responsibility of the calling code to make sure the object only gets updated when actual data
+      // changes
+      if (filtersEqual && nextProps.data === prevState.data) {
+        return {};
+      }
+      countsAndDuplicates = RuleTable.calculateCountAndDuplicates(nextProps.rules, nextProps.data);
+    } catch (e) {
+      // make sure to update state when checks/calculation fails
+    }
+    return {
+      data: nextProps.data,
+      rules: nextProps.rules,
+      countsAndDuplicates
+    };
+  }
+
+  static calculateDuplicates(matches: any[][]): number[] {
```

Comment:
I think this could be moved to the Util.

Reply:
Done in a49449f